internal/resources/resource_image.go

```diff
@@ -0,0 +1,112 @@
+// (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+
+package resources
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hewlettpackard/hpegl-metal-terraform-resources/pkg/client"
+)
+
+const iServiceImageFile = "os_service_image_file"
+
+func serviceSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		iServiceImageFile: {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Path to the YAML file containing the service image definition ",
+		},
+	}
+}
+
+func ServiceImageResource() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMetalImageCreate,
+		Read:   resourceMetalImageRead,
+		Delete: resourceMetalImageDelete,
+		Update: resourceMetalImageUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema:      serviceSchema(),
+		Description: "Provides service image resource. This allows creation, deletion and update of Metal OS service images .",
+	}
+}
+
+func resourceMetalImageCreate(d *schema.ResourceData, meta interface{}) (err error) {
+	defer wrapResourceError(&err, "failed to create OS service image")
+
+	filePath := safeString(d.Get(iServiceImageFile))
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		return fmt.Errorf("failed to open file %s, %v", filePath, err)
+	}
+
+	p, err := client.GetClientFromMetaMap(meta)
+	if err != nil {
+		return fmt.Errorf("failed to get client, %v", err)
```

Comment:
Same thing else where in this PR. Remove the "failed to..." in returned error.
```suggestion
		return fmt.Errorf("get client, %v", err)
```